tests/epyccel/test_epyccel_classes.py

```diff
@@ -40,3 +44,167 @@ def test_class_argument(modnew):
     assert x == 3.0
     x = modnew.get_an_x_from_B()
     assert x == -2.0
+
+def test_class_function(modnew):
+    c = modnew.C()
+    assert c.get_3() == 3
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [
+            pytest.mark.skip(reason="Argument reference count not increased. See #1586"),
+            pytest.mark.fortran]
+        ),
+        pytest.param("c", marks = [
+            pytest.mark.skip(reason="Argument reference count not increased. See #1586"),
```

Comment:
Did you mean #1686?

Reply:
#1586 is the issue. #1686 is the associated PR

Reply:
Currently #1686 doesn't actually fix this issue. The problem arises when a class stores a pointer to an object passed as argument. #1586 describes this problem. #1686 allows us to detect this however it then additionally needs to be taken into account in the wrapper to increase the reference count on the object and decrease it when the class is freed. I plan to fix this either in this PR or #1686 whichever is merged last, but I can create a separate PR if you think this will be clearer

Reply:
I think it is fine to fix it in #1686